airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java

```diff
@@ -213,6 +213,11 @@ public void close() throws Exception {
     }
   }
 
+  @Override
+  public Optional<AirbyteStreamNameNamespacePair> getAirbyteStream() {
```

Comment:
What is the problem/ticket resulting from the lack of this implementation?

Reply:
The default implementation of `getAirbyteStream()` is to return `Optional.empty()` so each iterator that implements the `AirbyteStreamAware` interface needs to implement this method. 

I noticed this during testing that we were throwing a NoSuchElementException instead of the relevant TransientException